src/stores/game.js

```diff
@@ -160,7 +166,10 @@ export const useGameStore = defineStore('game', {
       return state.turn % 2 === state.myPNum;
     },
     hasGlassesEight: (state) => {
-      return state.player.faceCards.filter((card) => card.rank === 8).length > 0;
+      if (!state.player) {
+        return false;
+      }
+      return state.player.faceCards?.filter((card) => card.rank === 8).length > 0 ?? false;
```

Comment:
I think these can be collapsed by just nullish coalescing state.players too like
```suggestion
      return state.player?.faceCards?.filter((card) => card.rank === 8).length > 0 ?? false;
```